projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -142,16 +142,24 @@ export class TableWidgetRendererComponent
       ])
     );
 
-    this.filterItems = this.model.getFilterOptions().map(filterOption => ({
-      label: capitalize(filterOption.label),
-      value: filterOption
-    }));
-
     this.viewItems = this.model.getViewOptions().map(viewOption => ({
       label: capitalize(viewOption),
       value: viewOption
     }));
 
+    this.populateStaticControls();
```

Comment:
this delineation between static and dynamic controls seems like we're embedding too much knowledge of the controls into the renderer.

Reply:
Yeah, open to suggestions then. Preferably without going down a rabbit hole. :-)

Reply:
of course, of course. When using on init to set the delegate, does this need go away?

Reply:
Hmm. Not really since there are still some controls that fetch data and some that don't.

We could just always repopulate all controls on fetch so the renderer doesn't have to have knowledge of which filters are static vs dynamic, but that seems worse. Not sure how else to handle this since we kind of mix and match how we provide data to our control filters. I kind of think they should all just use the modelAPI to getData(), even if its just an observable that returns a static list.

Reply:
> We could just always repopulate all controls on fetch so the renderer doesn't have to have knowledge of which filters are static vs dynamic, but that seems worse

Worse in that its an extra control render that's not necessary? If it's not noticeable, that seems preferable to me (not sure if that would cause things like losing control state on refresh, though).

>  I kind of think they should all just use the modelAPI to getData(), even if its just an observable that returns a static list.

Agreed, that way the caller doesn't know/care how the callee is implemented. Is this different than the "repopulate all on fetch" above?

Reply:
Yeah, okay. I think we agree. The populate on fetch does seem the least bad.